engine/runtime-integration-tests/src/test/java/org/enso/compiler/dump/test/DocsGenerateTest.java

```diff
@@ -374,6 +374,35 @@ public void intersectionTypes() throws Exception {
         sig);
   }
 
+  @Test
+  public void typesWithError() throws Exception {
+    var code =
+        """
+        type A
+        type B
+        type C
+
+        one a:A -> A & B ! C = a
+        """;
+
+    var v = new MockVisitor();
+    generateDocumentation("Error", code, v);
+
+    assertEquals("One methods", 1, v.visitMethod.size());
+    assertEquals("No constructors", 0, v.visitConstructor.size());
+
+    var p = v.visitMethod.get(0);
+    assertNull("It is a module method", p.t());
+
+    var m = p.ir();
+    var sig = DocsVisit.toSignature(m);
+    assertEquals("one", m.methodName().name());
+    assertEquals(
+        "Generates vector with argument type as return type",
+        "one a:local.Error.Main.A -> (local.Error.Main.A&local.Error.Main.B)!local.Error.Main.C",
```

Comment:
The core change in this PR. Signature `A & B ! C` gets recorded in the output with all three types mentioned.